non-atomic-action-interpreter-module/README.md

```diff
@@ -0,0 +1,122 @@
+# Non-Atomic Action Interpreter Module
+
+This module is part of the ostis-ps-lib and is responsible for interpreting non-atomic actions within the OSTIS.
+
+## Using module as an Extension for sc-machine
+
+The non-atomic action interpreter Module can be used as an extension to the sc-machine. Follow these steps to integrate it:
+
+### 1. Download sc-machine Artifacts
+
+   - Download pre-built artifacts of sc-machine from [GitHub Releases](https://github.com/ostis-ai/sc-machine/releases) and extract them to a location of your choice.
+   - Alternatively, you can build sc-machine from source and use the resulting artifacts.
+   - For more details on installing and using sc-machine artifacts, refer to the [sc-machine quick start guide](https://ostis-ai.github.io/sc-machine/quick_start/).
+
+### 2. Specify Extension Paths
+
+   - When running the sc-machine binary, specify the path to the `lib/extensions` directory from the extracted Non-Atomic Action Interpreter Module folder:
+
+   ```sh
+   ./path/to/sc-machine/binary -s path/to/kb.bin \
+       -e "non-atomic-action-interpreter-module/build/<Debug|Release>/lib/extensions;path/to/sc-machine/lib/extensions"
+   ```
+
+   Replace `path/to/sc-machine/binary` with the path to the sc-machine binary, and `path/to/kb.bin` with the path to your knowledge base file.
+
+## Developing Module
+
+### Installation Prerequisites
+
+Before you begin, ensure you have the following installed:
+
+- CMake (version 3.24 or higher);
+- C++ Compiler with C++17 support;
+- Conan package manager;
+- ccache (for faster rebuilds).
+
+!!! Note
+    This project is not supported on Windows OS.
+
+### Installation & Build
+
+#### 1. Clone Repository
+
+First, clone the repository containing the Non-Atomic Action Interpreter Module:
+
+```bash
+git clone https://github.com/ostis-ai/ostis-ps-lib.git
+cd ostis-ps-lib/non-atomic-action-interpreter-module
+```
+
+#### 2. Install Dependencies with Conan
+
+The project uses Conan to manage dependencies. Install the required dependencies:
+
+```bash
+conan install . --build=missing
+```
+
+#### 3. Configure Project
+
+You can configure the project using CMake presets. There are three main configuration options:
+
+- Debug with tests:
+  
+  ```sh
+  cmake --preset debug-conan
```

Comment:
this one failed
![image](https://github.com/user-attachments/assets/eba0afe6-30ec-4d70-9f3d-48edf059b40a)